karapace/schema_registry_apis.py

```diff
@@ -1055,7 +1055,7 @@ def _validate_references(
                 content_type=content_type,
                 status=HTTPStatus.BAD_REQUEST,
             )
-        if references and schema_type != SchemaType.PROTOBUF:
+        if references and schema_type != SchemaType.PROTOBUF and schema_type != SchemaType.AVRO:
```

Comment:
This condition is wrong, we will never have both schema types within the same request, thus this will not work.

Reply:
Let me take another look, because I see you also use the API client in the integration tests and if the tests pass then this should be correct.

Reply:
I guess the two `and` bugs me out, maybe a check for inclusion, i.e. `if references and (schema_type not in {SchemaType.PROTOBUF, SchemaType.AVRO})`, but it's also fine as the tests confirm. 
I think when I was testing the API via postman against the docker compose of your fork, the request might have still been hitting the docker compose I had running via the upstream.